extensions/pip/main.ts

```diff
@@ -0,0 +1,214 @@
+      "~/.pyenv",
+      "/tmp",
+      "/etc/passwd",
+      "/etc/apache2/mime.types",
+    ],
+    net: true,
+  },
+});
+Deno.exit(installStatus.code ?? 255);
+
+type JobStatus = {
+  packages: {
+    issues: {
+      severity: string;
+      title: string;
+    }[];
+  }[];
+};
+
+// Logs any identified issues to the screen.
+function logIssues(jobStatus: Record<string, unknown>) {
+  for (const pkg of (jobStatus as JobStatus).packages) {
+    const issues = pkg.issues;
+
+    for (const issue of issues) {
+      let severity = issue.severity.toLowerCase();
+
+      if (["high", "critical"].indexOf(severity) != -1) {
+        severity = red(severity);
+      } else if (severity == "medium") {
+        severity = yellow(severity);
+      } else {
+        severity = green(severity);
+      }
+
+      console.log(`    [${severity}] ${issue.title}`);
+    }
+  }
+}
+
+// Analyze new packages.
+async function checkDryRun() {
+  console.log(`[${green("phylum")}] Finding new dependencies…`);
+
+  const status = PhylumApi.runSandboxed({
+    cmd: "pip3",
+    args: [...Deno.args, "--quiet", "--report", "-", "--dry-run"],
+    exceptions: {
+      run: ["./", "/bin", "/usr/bin", "/usr/local/bin", "~/.pyenv"],
+      write: [
+        "~/Library/Caches",
+        "~/.pyenv",
+        "~/.cache",
+        "~/.local/lib",
+        "/tmp",
+      ],
+      read: [
+        "~/Library/Caches",
+        "~/.cache",
+        "~/.local/lib",
+        "/tmp",
+        "/etc/passwd",
+        "/etc/apache2/mime.types",
+      ],
+      net: true,
+    },
+    stdout: "piped",
+  });
+
+  // Ensure dry-run was successful.
+  if (!status.success) {
+    console.error(`[${red("phylum")}] Pip dry-run failed.\n`);
+    Deno.exit(status.code ?? 255);
+  }
+
+  // Parse dry-run output.
+  const packages = parseDryRun(status.stdout);
+
+  console.log(`[${green("phylum")}] Dependency resolution successful.\n`);
+  console.log(`[${green("phylum")}] Analyzing packages…`);
+
+  if (packages.length === 0) {
+    console.log(`[${green("phylum")}] No new packages found for analysis.\n`);
+    return;
+  }
+
+  const jobId = await PhylumApi.analyze("pypi", packages);
+  const jobStatus = await PhylumApi.getJobStatus(jobId);
+
+  if (jobStatus.pass && jobStatus.status === "complete") {
+    console.log(`[${green("phylum")}] All packages pass project thresholds.\n`);
+  } else if (jobStatus.pass) {
+    console.warn(
+      `[${
+        yellow(
+          "phylum",
+        )
+      }] Unknown packages were submitted for analysis, please check again later.\n`,
+    );
+    Deno.exit(126);
+  } else {
+    console.error(
+      `[${red("phylum")}] The operation caused a threshold failure.\n`,
+    );
+
+    logIssues(jobStatus);
+    Deno.exit(127);
+  }
+}
+
+type Package = {
+  name: string;
+  version: string;
+  package_type: string;
+};
+
+type DryRunReport = {
+  install: {
+    metadata: {
+      name: string;
+      version: string;
+    };
+  }[];
+};
```

Comment:
The [pip installation report specification](https://pip.pypa.io/en/stable/reference/installation-report/) says:
> version: the string 1. It will change only if and when backward incompatible changes are introduced, such as removing mandatory fields or changing the semantics or data type of existing fields. The introduction of backward incompatible changes will follow the usual pip processes such as the deprecation cycle or feature flags. Tools must check this field to ensure they support the corresponding version.

The fields expected to exist in this report are limited and therefore the chance of breaking changes is reduced, but even still, should there be such a check in this extension?